lib/ui5Framework/maven/Installer.js

```diff
@@ -0,0 +1,505 @@
+import path from "node:path";
+import {mkdirp} from "../../utils/fs.js";
+import fs from "graceful-fs";
+import _StreamZip from "node-stream-zip";
+const StreamZip = _StreamZip.async;
+import {promisify} from "node:util";
+import Registry from "./Registry.js";
+import AbstractInstaller from "../AbstractInstaller.js";
+import rimraf from "rimraf";
+const stat = promisify(fs.stat);
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+const rename = promisify(fs.rename);
+const rm = promisify(fs.rm);
+import {getLogger} from "@ui5/logger";
+const log = getLogger("ui5Framework:maven:Installer");
+const mvnTimestampRegex = /^(\d{4})(\d\d)(\d\d)(\d\d)(\d\d)(\d\d)$/;
+
+const CACHE_TIME = 32400000; // 9 hours
+
+class Installer extends AbstractInstaller {
+	/**
+	 * @param {object} parameters Parameters
+	 * @param {string} parameters.ui5HomeDir UI5 home directory location. This will be used to store packages,
+	 * metadata and configuration used by the resolvers.
+	 * @param {Promise<string>} parameters.snapshotEndpointUrl Promise, resolving to Maven repository URL.
```

Comment:
Fully agree with that and that was my idea initially. Please take a look at this commit: https://github.com/SAP/ui5-project/pull/570/commits/3bd183b6cb94390b5c2270fb48266dd6830a9827

It works perfectly fine unless you'd like to work in "offline" mode. The issue is that in that case, although the artifacts and packages are available locally, the end user would still face the prompt with the question of whether to resolve the URL from settings.xml. 
The core reasoning is that the cache logic is inside `maven/Registry`, inside `maven/Installer`. Basically, the earliest possible moment to "know" about the cache is inside the `maven/Installer`

Reply:
Understood.
So after looking into the code again `parameters.snapshotEndpointUrl` is not of type `Promise<string>`. It is an async callback that resolves with the URL, which makes sense to support the described behavior.
I was confused by the documented type as a Promise gives the installer no way to tell when the result is needed. It could only wait for it.

Please adapt the parameter name to make clear that it is an async callback that returns the snapshotEndpointUrl.

Reply:
Done